lib/cookie.ts

```diff
@@ -1247,64 +1307,78 @@ export function createPromiseCallback<T>(args: IArguments): PromiseCallback<T> {
     reject = _reject
   })
 
-  if (typeof args[args.length - 1] === 'function') {
-    const cb = args[args.length - 1]
+  const cb = args[args.length - 1] as unknown
+  if (typeof cb === 'function') {
     callback = (err, result) => {
       try {
         cb(err, result)
       } catch (e) {
-        reject(e instanceof Error ? e : new Error(`${e}`))
+        reject(e instanceof Error ? e : new Error())
```

Comment:
Why this change? We should probably have some kind of message to be a bit more helpful, and maybe include the original error. (Also returning so we safely short-circuit.)
```suggestion
        if (e instanceof Error) return reject(e)
        if (typeof e !== 'object' || e === null) return reject new Error(e)
        const wrappedError = new Error('An unknown error occurred.')
        wrappedError.original = e
        return reject(wrappedError)
```